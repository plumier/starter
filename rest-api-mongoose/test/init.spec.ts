import "@plumier/testing"

import dotenv from "dotenv"
import mongoose from "mongoose"
import { join } from "path"

import createApp from "../src/app"

dotenv.config({ path: join(__dirname, "./.env-test") })


it("Should initialized properly", async () => {
    const mock = console.mock()
    const app = await createApp()
    console.mockClear()
    expect(mock.mock.calls).toMatchSnapshot()
    await mongoose.disconnect()
})
