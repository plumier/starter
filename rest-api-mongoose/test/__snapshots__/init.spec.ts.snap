// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should initialized properly 1`] = `
Array [
  Array [],
  Array [
    "Route Analysis Report",
  ],
  Array [
    "1. AuthController.login(email, password) -> Public        POST   /auth/login",
  ],
  Array [
    "2. AuthController.refresh(user)          -> RefreshToken  POST   /auth/refresh",
  ],
  Array [
    "3. AuthController.logout()               -> Private       POST   /auth/logout",
  ],
  Array [
    "4. MongooseControllerGeneric.get         -> ResourceOwner GET    /users/:id",
  ],
  Array [
    "5. MongooseControllerGeneric.modify      -> ResourceOwner PATCH  /users/:id",
  ],
  Array [
    "6. MongooseControllerGeneric.replace     -> ResourceOwner PUT    /users/:id",
  ],
  Array [
    "7. MongooseControllerGeneric.delete      -> ResourceOwner DELETE /users/:id",
  ],
  Array [
    "8. MongooseControllerGeneric.list        -> Admin         GET    /users",
  ],
  Array [
    "9. MongooseControllerGeneric.save        -> Public        POST   /users",
  ],
  Array [],
]
`;
